src/pybamm/expression_tree/operations/serialise.py

```diff
@@ -399,3 +761,303 @@ def _convert_options(self, d):
             return tuple(self._convert_options(item) for item in d)
         else:
             return d
+
+    @staticmethod
+    def convert_symbol_to_json(symbol: pybamm.Symbol) -> dict[str, Any]:
+        """
+        Recursively converts a PyBaMM symbolic expression into a JSON-serializable format.
+
+        Supports all major PyBaMM symbol types, including :class:`pybamm.Scalar`,
+        :class:`pybamm.Variable`, :class:`pybamm.Parameter`, :class:`pybamm.Operator`,
+        :class:`pybamm.FunctionParameter`, :class:`pybamm.Broadcast`, and
+        :class:`pybamm.Interpolant`.
+
+        Parameters
+        ----------
+        symbol : pybamm.Symbol or compatible type
+            The expression or object to convert.
+
+        Returns
+        -------
+        dict
+            A JSON-compatible representation of the input.
+
+        Examples
+        --------
+        >>> import pybamm
+        >>> from pybamm.expression_tree.operations.serialise import Serialise
+        >>> s = pybamm.Scalar(5)
+        >>> Serialise.convert_symbol_to_json(s)
+        {'type': 'Scalar', 'value': 5.0}
+
+        >>> v = pybamm.Variable("c")
+        >>> Serialise.convert_symbol_to_json(v)
+        {'type': 'Variable', 'name': 'c', 'domains': {'primary': [], 'secondary': [], 'tertiary': [], 'quaternary': []}, 'bounds': [{'type': 'Scalar', 'value': -inf}, {'type': 'Scalar', 'value': inf}]}
+        """
+
+        if isinstance(symbol, numbers.Number | list):
+            return symbol
```

Comment:
The function accepts `numbers.Number` and `list` objects as well, could you please update the type hint as well? i.e., see my other comment